dune/dune.go

```diff
@@ -167,7 +173,45 @@ func (c *duneClient) QueryResults(executionID string) (*models.ResultsResponse,
 }
 
 func (c *duneClient) QueryResultsCSV(executionID string) (io.Reader, error) {
-	resultsURL := fmt.Sprintf(resultsCSVURLTemplate, c.env.Host, executionID)
+	resultsURL := fmt.Sprintf(executionResultsCSVURLTemplate, c.env.Host, executionID)
+	req, err := http.NewRequest("GET", resultsURL, nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := httpRequest(c.env.APIKey, req)
+	if err != nil {
+		return nil, err
+	}
+
+	// we read whole result into ram here. if there was a paginated API we wouldn't need to
+	var buf bytes.Buffer
+	defer resp.Body.Close()
+	_, err = buf.ReadFrom(resp.Body)
+	return &buf, err
+}
+
+func (c *duneClient) QueryResultsByQueryID(queryID string) (*models.ResultsResponse, error) {
+	resultsURL := fmt.Sprintf(queryResultsURLTemplate, c.env.Host, queryID)
```

Comment:
That is the only line that is different from QueryResults func.
How about creating a function `func (c *duneClient) getQueryResults(url string) (*models.ResultsResponse, error)`, and call it from QueryResultsByQueryID and QueryResults?

The same comment is valid for QueryResultsCSV and QueryResultsCSVByQueryID.

Reply:
yeah, fair :smile: